pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1595,241 +1621,120 @@ func parseConfigItems(config *zconfig.EdgeDevConfig, ctx *getconfigContext) {
 	// Start with the defaults so that we revert to default when no data
 	newGlobalConfig := types.GlobalConfigDefaults
 
+	// Reset ConfigItem Errors
+	globalStatusPtr := &ctx.zedagentCtx.globalStatus
+	globalStatusPtr.Errors = make([]string, 0)
+
 	errStr := ""
 	for _, item := range items {
 		log.Infof("parseConfigItems key %s value %s\n",
 			item.Key, item.Value)
-		if errStr != "" {
-			newGlobalConfig.Errors = append(newGlobalConfig.Errors, errStr)
-			errStr = ""
-		}
-
 		key := item.Key
 		switch key {
 		case "timer.config.interval":
-			i64, err := strconv.ParseInt(item.Value, 10, 32)
-			if err != nil {
-				log.Errorf("parseConfigItems: bad int value %s for %s: %s\n",
-					item.Value, key, err)
-				continue
-			}
-			newGlobalConfig.ConfigInterval = uint32(i64)
+			errStr = parseU32ConfigItem(&newGlobalConfig, key,
```

Comment:
This isn't needed if you use the correct way to pass errors.
Just replace the original continue with an else clause.

Reply:
Yeah.. but at that point, it was a lot of repetitive pattern.. And hence had this function.
Now that I made the change, I think this is better anyways.. isn't it?

Reply:
I actually find it harder to understand the new code.
Having return values is a lot clearer than functions which modify pass-by-value arguments.
And since there are just parsing functions it makes sense to me to have them use the same calling/return approach as ParseInt() and friends.